novelist.py

```diff
@@ -0,0 +1,187 @@
+            raise Exception("Invalid NoveList credentials")
+
+        return self.lookup_info_to_metadata(representation.content)
+
+    @classmethod
+    def _build_query(cls, params):
+        """Builds a unique and url-encoded query endpoint"""
+
+        for name, value in params.items():
+            params[name] = urllib.quote(value)
+        return (cls.QUERY_ENDPOINT % params).replace(" ", "")
+
+    def lookup_info_to_metadata(self, lookup_info):
+        """Turns a NoveList JSON response into a Metadata object"""
+
+        lookup_info = json.loads(lookup_info)
+        
+        urn = urllib.unquote(lookup_info['ClientIdentifier'])
+        primary_identifier, ignore = Identifier.parse_urn(self._db, urn)
+        metadata = Metadata(
+            self._db, self.source, primary_identifier=primary_identifier
+        )
+
+        # Get the equivalent ISBN identifiers.
+        book_info = lookup_info['TitleInfo']
+        synonymous_ids = book_info.get('manifestations')
+        for synonymous_id in synonymous_ids:
+            isbn = synonymous_id.get('ISBN')
+            if isbn and isbn != primary_identifier.identifier:
+                isbn_data = IdentifierData(Identifier.ISBN, isbn)
+                metadata.identifiers.append(isbn_data)
+
+        metadata.title = book_info.get('main_title')
+        metadata.subtitle = self._subtitle(metadata.title, book_info.get('full_title'))
+
+        author = book_info.get('author')
+        if author:
+            metadata.contributors.append(ContributorData(sort_name=author))
+
+        description = book_info.get('description')
+        if description:
+            metadata.links.append(LinkData(
+                rel=Hyperlink.DESCRIPTION, content=description,
+                media_type=Representation.TEXT_PLAIN
+            ))
+
+        audience_level = book_info.get('audience_level')
+        if audience_level:
+            metadata.subjects.append(SubjectData(
+                Subject.FREEFORM_AUDIENCE, audience_level
+            ))
+
+        novelist_rating = book_info.get('rating')
+        if novelist_rating:
+            metadata.measurements.append(MeasurementData(
+                Measurement.RATING, novelist_rating
+            ))
+
+        # Extract feature content if it is available.
+        lexile_info = series_info = goodreads_info = appeals_info = None
+        feature_content = lookup_info.get('FeatureContent')
+        if feature_content:
+            lexile_info = feature_content.get('LexileInfo')
+            series_info = feature_content.get('SeriesInfo')
+            goodreads_info = feature_content.get('GoodReads')
+            appeals_info = feature_content.get('Appeals')
+
+        if series_info:
+            metadata.series = series_info['full_title']
+
+        if appeals_info:
+            extracted_genres = False
+            for appeal in appeals_info:
+                genres = appeal.get('genres')
+                if genres:
```

Comment:
What are the genres? Is there a small enough number of genres that it's worth creating a new Subject type with its own Classifier, rather than using `tag`?